adcm_aio_client/core/accessors.py

```diff
@@ -9,35 +9,170 @@
+            page = 1
+        if items is None:
+            items = 10
+
+        current_page = page
+
+        while True:
+            # Calculate the start and end indices for the current page
+            start_index = (current_page - 1) * items
+            end_index = current_page * items
+            yield (start_index, end_index)
+            current_page += 1
+
+    async def create(self: Self, **kwargs) -> T:
+        # Simulate a request that creates a new object
+        response: RequesterResponse = await MockRequester.post(self.path, kwargs)
+        assert response.as_dict()["status_code"] == 201  # TODO: rework accoring to actual API response structure
+        return self._create_object(kwargs)
+
+    async def get(self, **predicate) -> T:
+        if predicate:
+            filter_objects = Filter(**predicate)
+
+            if not filter_objects.is_valid():
+                raise MissingParameterException("Filter parameters are missing or invalid.")
+
+        # Simulate a request that returns objects based on a filter
+        response: RequesterResponse = await MockRequester.get(self.path, self.query_params)
+        objects = response.as_dict()
+
+        if not objects:
+            raise ObjectDoesNotExistError("No objects found with the given filter.")
+        elif len(objects) > 1:
+            raise MultipleObjectsReturnedError("More than one object found.")
+        else:
+            return self._create_object(objects[0])
+
+    async def list(self: Self, offset: int = 1, limit: int = 10) -> List[T]:
+        response: RequesterResponse = await MockRequester.list(
+            self.path, self.query_param.update({"offset": offset, "limit": limit})
+        )
+        return [self._create_object(object) for object in response.as_list()]
+
+    async def get_or_none(self: Self, predicate: T) -> T | None:
+        with suppress(ObjectDoesNotExistError):
+            obj = await self.get(predicate=predicate)
+            if obj:
+                yield obj
+            else:
+                yield None
+        yield None  # Default yield in case of exceptions
+
+    async def all(self: Self, page: int = 1, items: int = 10) -> List[T]:
+        objects = []
+        try:
+            while True:
+                batch = await self.list(page, items)
+                objects.append(batch)
+                page += 1
+        except StopIteration:
+            return objects
+
+    async def iter(self: Self, offset: int = 1, limit: int = 10) -> AsyncGenerator[List[T]]:
+        paginator = self.paginate(offset, limit)
+        while True:
+            start, end = next(paginator)
+            response: RequesterResponse = await MockRequester.list(
+                self.path, self.query_param.update({"offset": start, "limit": end})
+            )
+            if start >= len(response.as_list()):
+                raise StopIteration
+            yield [self._create_object(object) for object in response.as_list()]
```

Comment:
same as for the all, also should yield exact objects, not lists of them

I believe all should just `return list(self.iter())`